services/minespace-web/src/components/dashboard/mine/permits/PermitsTable.js

```diff
@@ -1,33 +1,157 @@
 import React from "react";
-import { Table } from "antd";
+import { Table, Typography } from "antd";
+import { connect } from "react-redux";
 import PropTypes from "prop-types";
+import { formatDate } from "@common/utils/helpers";
+import moment from "moment";
+import { getDropdownPermitStatusOptions } from "@common/selectors/staticContentSelectors";
+import { downloadFileFromDocumentManager } from "@common/utils/actionlessNetworkCalls";
+import CustomPropTypes from "@/customPropTypes";
+import LinkButton from "@/components/common/LinkButton";
+import * as Strings from "@/constants/strings";
+
+const { Text } = Typography;
 
 const propTypes = {
   isLoaded: PropTypes.bool.isRequired,
+  permits: PropTypes.arrayOf(CustomPropTypes.permit).isRequired,
+  permitStatusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
 };
 
 const columns = [
-  { title: "Permit No.", dataIndex: "permit_no", key: "permit_no", sorter: true },
-  { title: "Permit Status", dataIndex: "permit_status", key: "permit_status", sorter: true },
-  { title: "Permit End Date", dataIndex: "permit_end_date", key: "permit_end_date", sorter: true },
-  { title: "First Issued", dataIndex: "first_issued", key: "first_issued", sorter: true },
-  { title: "Last Amended", dataIndex: "last_amended", key: "last_amended", sorter: true },
+  {
+    title: "Permit No.",
+    dataIndex: "number",
+    key: "number",
+    sorter: (a, b) => (a.number > b.number ? -1 : 1),
+  },
+  {
+    title: "Permit Status",
+    dataIndex: "status",
+    key: "status",
+    sorter: (a, b) => (a.status > b.status ? -1 : 1),
+  },
+  {
+    title: "Authorization End Date",
+    dataIndex: "authorizationEndDate",
+    key: "authorizationEndDate",
+    sorter: (a, b) => (moment(a.authorizationEndDate) > moment(b.authorizationEndDate) ? -1 : 1),
+  },
+  {
+    title: "First Issued",
+    dataIndex: "firstIssued",
+    key: "firstIssued",
+    sorter: (a, b) => (moment(a.firstIssued) > moment(b.firstIssued) ? -1 : 1),
+  },
+  {
+    title: "Last Amended",
+    dataIndex: "lastAmended",
+    key: "lastAmended",
+    sorter: (a, b) => (moment(a.lastAmended) > moment(b.lastAmended) ? -1 : 1),
+  },
 ];
 
-const data = [];
-
-export const PermitsTable = (props) => (
-  <Table
-    size="small"
-    pagination={false}
-    loading={!props.isLoaded}
-    columns={columns}
-    expandedRowRender={(record) => <p>{record.description}</p>}
-    dataSource={data}
-    locale={{ emptyText: "This mine has no permit data." }}
-  />
+const renderDocumentLink = (file, text) => (
```

Comment:
Does this being a function provide value and it seems like this is only used once? Can't expandedColumn include a <LinkButton directly?